cloudify_aws/eks/resources/cluster.py

```diff
@@ -0,0 +1,227 @@
+            encoded.decode('utf-8').rstrip('=')
+        # build the cluster config hash
+        cluster_config = {
+            "apiVersion": "v1",
+            "kind": "Config",
+            "clusters": [
+                {
+                    "cluster": {
+                        "server": str(cluster_ep),
+                        "certificate-authority-data": str(cluster_cert)
+                    },
+                    "name": "kubernetes"
+                }
+            ],
+            "contexts": [
+                {
+                    "context": {
+                        "cluster": "kubernetes",
+                        "user": "aws"
+                    },
+                    "name": "aws"
+                }
+            ],
+            "current-context": "aws",
+            "preferences": {},
+            "users": [
+                {
+                    "name": "aws",
+                    "user": {
+                        "token": token
+                    }
+                }
+            ]
+        }
+        return cluster_config
+
+    def delete(self, params=None):
+        """
+            Deletes an existing AWS EKS cluster.
+        """
+        res = self.client.delete_cluster(
+            **{CLUSTER_NAME: params.get(CLUSTER_NAME)}
+        )
+        self.logger.debug('Response: {}'.format(res))
+        return res
+
+
+def prepare_describe_cluster_filter(params, iface):
+    iface.describe_param = {
+        CLUSTER_NAME: params.get(CLUSTER_NAME),
+    }
+    return iface
+
+
+@decorators.aws_resource(resource_type=RESOURCE_TYPE)
+def prepare(ctx, resource_config, **_):
+    """Prepares an AWS EKS Cluster"""
+    # Save the parameters
+    ctx.instance.runtime_properties['resource_config'] = resource_config
+
+
+@decorators.aws_resource(EKSCluster, RESOURCE_TYPE)
+def create(ctx, iface, resource_config, **_):
+    """Creates an AWS EKS Cluster"""
+    store_kube_config_in_runtime = \
+        ctx.node.properties['store_kube_config_in_runtime']
+    params = dict() if not resource_config else resource_config.copy()
+    resource_id = \
+        utils.get_resource_id(
+            ctx.node,
+            ctx.instance,
+            params.get(CLUSTER_NAME),
+            use_instance_id=True
+        )
+
+    utils.update_resource_id(ctx.instance, resource_id)
+    iface = prepare_describe_cluster_filter(resource_config.copy(), iface)
+    response = iface.create(params)
+    if response and response.get(CLUSTER):
+        resource_arn = response.get(CLUSTER).get(CLUSTER_ARN)
+        utils.update_resource_arn(ctx.instance, resource_arn)
+    # wait for cluster to be active
+    ctx.logger.info("Waiting for Cluster to become Active")
+    iface.wait_for_cluster(params, 'cluster_active')
+    if store_kube_config_in_runtime:
+        client_config = ctx.node.properties['client_config']
+        ctx.instance.runtime_properties['kubeconf'] = \
+            iface.get_kubeconf(client_config, params)
```

Comment:
Is there anychance that this could contain some object that is not serializable in JSON?